app/domain/authentication/authn_jwt/id_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,58 @@
+module Authentication
+  module AuthnJwt
+    # Class for providing jwt identity from the decoded token from the field specified in a secret
+    class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+
+      attr_accessor :secret_fetcher
+
+      def initialize(authentication_parameters)
+        @resource_id = authentication_parameters.authenticator_resource_id
+        @decoded_token = authentication_parameters.decoded_token
+        @secret_fetcher = Conjur::FetchRequiredSecrets
+      end
+
+      def provide_jwt_identity
+        token_field_name = fetch_token_field_name
+        if @decoded_token.include?(token_field_name)
+          @decoded_token[token_field_name]
+        else
+          raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken.new(
+            token_field_name
+          )
+        end
```

Comment:
I know we already talked about it, but please consider better and more readable approach:
```suggestion
        jwt_identity = @decoded_token[token_field_name]
        if jwt_identity.blank?
          raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken.new(
            token_field_name
          )
        end
        jwt_identity
```

Reply:
Done